frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/os-flavor.tsx

```diff
@@ -91,28 +113,94 @@ export const OSFlavor: React.FC<OSFlavorProps> = React.memo(
+      (operatingSystem: OperatingSystemRecord) => {
+        const pvcName = operatingSystem?.dataVolumeName;
+        const baseImageFoundInCluster = loadedBaseImages?.find((pvc) => iGetName(pvc) === pvcName);
+        const osField = {
+          id: operatingSystem.id,
+          name: operatingSystem.name,
+          pvcName,
+          baseImageFoundInCluster,
+          message: '',
+          longMessage: '',
+        };
+
+        if (!userTemplate) {
+          if (baseImageFoundInCluster && pvcName) {
+            osField.message = BASE_IMAGE_AND_PVC_SHORT;
+            osField.longMessage = BASE_IMAGE_AND_PVC_MESSAGE;
+          } else if (pvcName) {
+            osField.message = NO_BASE_IMAGE_SHORT;
+            osField.longMessage = NO_BASE_IMAGE_MESSAGE;
+          } else {
+            osField.message = NO_BASE_IMAGE_AND_NO_PVC_SHORT;
+            osField.longMessage = NO_BASE_IMAGE_AND_NO_PVC_MESSAGE;
+          }
+        }
+
+        return osField;
+      },
+    );
+    const baseImage = operatingSystemBaseImages.find((image) => image.id === os);
+
     return (
       <>
-        <FormFieldRow
-          field={operatinSystemField}
-          fieldType={FormFieldType.SELECT}
-          validation={operatingSystemValidation}
-          loadingResources={loadingResources}
-        >
-          <FormField value={displayOnly ? display : undefined}>
-            <FormSelect onChange={nullOnEmptyChange(onChange, VMSettingsField.OPERATING_SYSTEM)}>
-              {!displayOnly && (
-                <FormSelectPlaceholderOption
-                  placeholder={getPlaceholder(VMSettingsField.OPERATING_SYSTEM)}
-                  isDisabled={!!os}
-                />
-              )}
-              {operatingSystems.map(({ id, name }) => {
-                return <FormSelectOption key={id} value={id} label={name || id} />;
-              })}
-            </FormSelect>
-          </FormField>
-        </FormFieldRow>
+        <div className="pf-c-form__group">
+          <FormFieldRow
+            field={operatinSystemField}
+            fieldType={FormFieldType.SELECT}
+            validation={operatingSystemValidation}
+            loadingResources={loadingResources}
+          >
+            <FormField value={displayOnly ? display : undefined}>
+              <FormSelect onChange={nullOnEmptyChange(onChange, VMSettingsField.OPERATING_SYSTEM)}>
+                {!displayOnly && (
+                  <FormSelectPlaceholderOption
+                    placeholder={getPlaceholder(VMSettingsField.OPERATING_SYSTEM)}
+                    isDisabled={!!os}
+                  />
+                )}
+                {operatingSystemBaseImages.map(({ id, name, message }) => (
+                  <FormSelectOption key={id} value={id} label={`${name || id} ${message}`} />
+                ))}
+              </FormSelect>
+            </FormField>
+          </FormFieldRow>
+          <FormFieldRow
```

Comment:
Can you take the checkbox out of this `pf-c-form__group` and put it in a separate one? I'm seeing nested groups in the rendered markup.  It also looks like  there is styling added via forms.scss that adds this
```
input[type="radio"], input[type="checkbox"] {
    margin: 1px 0 0;
    margin-top: 1px \9;
    /* line-height: normal; */
}
```
Which is messing up the alignment :(

Reply:
@matthewcarleton this is how it looks if the checkbox is not nested ..
I didn't like the spacing before the checkbox, but I'm ok with it too ...

![screenshot-localhost_9000-2020 07 27-15_55_16](https://user-images.githubusercontent.com/2181522/88544209-b658ea00-d021-11ea-828a-49994baa6e61.png)

Reply:
removed the unhealthy nesting